tests/PHPUnit/Framework/TestCase/SystemTestCase.php

```diff
@@ -287,6 +289,65 @@ protected static function getApiForTestingScheduledReports($dateTime, $period)
         return $apiCalls;
     }
 
+    /**
+     * While {@link runApiTests()} lets you run test for many API methods at once this one tests only one specific
+     * API method and it goes via HTTP. While the other method lets you test only some methods starting with 'get'
+     * this one lets you actually test any API method.
+     */
+    protected function runAnyApiTest($apiMethod, $apiId, $requestParams, $options = array())
```

Comment:
I kinda had to add this method here. I tried to add a system test with regular `runApiTests()` method but failed. My main problem was that I needed to execute system tests in different user contexts. Eg a user having only view or only admin access. I tried lots of different things but somehow it didn't work. Going actually via HTTP made it work immediately and this way we actually test proper check of `Access` etc.